ntlm3/compute_response.py

```diff
@@ -0,0 +1,375 @@
+        target_info = None
+
+        if self._negotiate_flags & NegotiateFlags.NTLMSSP_NEGOTIATE_EXTENDED_SESSIONSECURITY and self._ntlm_compatibility < 3:
+            # The compatibility level is less than 3 which means it doesn't support NTLMv2 but we want extended security so use NTLM2 which is different from NTLMv2
+            # [MS-NLMP] - 3.3.1 NTLMv1 Authentication
+            response = self._get_NTLM2_response(self._password, self._server_challenge, self._client_challenge)
+
+        elif 0 <= self._ntlm_compatibility < 3:
+            response = self._get_NTLMv1_response(self._password, self._server_challenge)
+        else:
+            # [MS-NLMP] - 3.3.2 NTLMv2 Authentication
+            if self._server_target_info is None:
+                target_info = TargetInfo()
+            else:
+                target_info = self._server_target_info
+
+            if target_info[TargetInfo.MSV_AV_TIMESTAMP] is None:
+                timestamp = self._get_windows_timestamp()
+            else:
+                timestamp = target_info[TargetInfo.MSV_AV_TIMESTAMP][1]
+                # TODO: Need to calculate the MIC when this flag is present
+
+            if server_certificate_hash != None:
+                channel_bindings_hash = self._get_channel_bindings_value(server_certificate_hash)
+                target_info[TargetInfo.MSV_AV_CHANNEL_BINDINGS] = channel_bindings_hash
+
+            response = self._get_NTLMv2_response(self._domain, self._user_name, self._password, self._server_challenge, self._client_challenge, timestamp, target_info)
+
+        return response, target_info
+
+
+    """
+        [MS-NLMP] v28.0 2016-07-14
+
+        3.3.1 NTLM v1 Authentication
+        Note: Same function as LMOWFv1 in document
+    """
+    def _lmowfv1(self, password):
+        """create LanManager hashed password"""
+
+        # if the password provided is already a hash, we just return the first half
+        if re.match(r'^[\w]{32}:[\w]{32}$', password):
+            return binascii.unhexlify(password.split(':')[0])
+
+        # fix the password length to 14 bytes
+        password = password.upper()
+        lm_pw = password + '\0' * (14 - len(password))
+        lm_pw = password[0:14]
+
+        # do hash
+        magic_str = b"KGS!@#$%"  # page 57 in [MS-NLMP]
+
+        res = b''
+        dobj = des.DES(lm_pw[0:7])
+        res = res + dobj.encrypt(magic_str)
+
+        dobj = des.DES(lm_pw[7:14])
+        res = res + dobj.encrypt(magic_str)
+
+        return res
+
+    """
+        [MS-NLMP] v28.0 2016-07-14
+
+        3.3.1 NTLM v1 Authentication
+        Note: Same function as NTOWFv1 in document
+    """
+    def _ntowfv1(self, password):
+        "create NT hashed password"
+        # if the password provided is already a hash, we just return the second half
+        if re.match(r'^[\w]{32}:[\w]{32}$', password):
+            return binascii.unhexlify(password.split(':')[1])
```

Comment:
Same as the comment in _lmowfv1, is this needed? If so can we write a test for it?